HasStyle.setClassName fails when null is passed

Vaaidn 14.3.2, flow 2.3.2. The following code:

```java
new Button("Foo").setClassName(null);
```

fails with

```
Caused by: java.lang.IllegalArgumentException: Value cannot be null
	at com.vaadin.flow.dom.Element.validateAttribute(Element.java:1448)
	at com.vaadin.flow.dom.Element.setAttribute(Element.java:286)
	at com.vaadin.flow.component.HasStyle.setClassName(HasStyle.java:66)
```

even though `HasStyle.setClassName()` javadoc says

```
     * @param className
     *            a space-separated string of class names to set, or
     *            <code>null</code> to remove all class names
```

--- a/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
+++ b/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
@@ -63,7 +63,11 @@ public interface HasStyle extends HasElement {
      *            <code>null</code> to remove all class names
      */
     default void setClassName(String className) {
-        getElement().setAttribute("class", className);
+        if(className == null) {
+            getElement().removeAttribute("class");
+        } else {
+            getElement().setAttribute("class", className);
+        }
     }
 
     /**

--- a/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java
@@ -70,6 +70,10 @@ public class HasStyleTest {
         component.removeClassName("baz");
         assertClasses(component);
 
+        component.setClassName("removeMe");
+        // setting null to classname should remove class name attribute
+        component.setClassName(null);
+
         // use ClassList
         component.setClassName("foo");
 

--- a/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java
@@ -70,10 +70,6 @@ public class HasStyleTest {
         component.removeClassName("baz");
         assertClasses(component);
 
-        component.setClassName("removeMe");
-        // setting null to classname should remove class name attribute
-        component.setClassName(null);
-
         // use ClassList
         component.setClassName("foo");
 
@@ -93,6 +89,11 @@ public class HasStyleTest {
         component.setClassName(" ");
         assertClasses(component);
         component.setClassName("");
+        assertClasses(component);
+
+        component.setClassName("removeMe");
+        // setting null to classname should remove class name attribute
+        component.setClassName(null);
         assertClasses(component);
     }
 

